core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,8 +321,11 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            if (deprecationLogger.get() == null) {
```

Comment:
@pickypg I think that's better too so I pushed 308923905d84346a13592347c4441308279ad08d.